auth/src/main/java/org/aerogear/mobile/security/AbstractSecurityCheckExecutor.java

```diff
@@ -31,50 +33,64 @@
     private final MetricsService metricsService;
 
     /**
-     * Constructor.
+     * Creates AbstractSecurityCheckExecutor object.
      *
-     * @param context the context
-     * @param checks checks to be executed
+     * @param context the context.
+     * @param checks checks to be executed.
      * @param metricService singleThreadService to be used to publish metrics. If null, no metrics get published.
+     * @throws IllegalArgumentException if context or checks are null.
      */
     public AbstractSecurityCheckExecutor(@NonNull final Context context,
                                          @NonNull final Collection<SecurityCheck> checks,
                                          @Nullable final MetricsService metricService) {
-        this.context = context;
-        this.metricsService = metricService;
+        this.context = nonNull(context, "context");
+        this.metricsService = nonNull(metricService, "metricService");
```

Comment:
I think `metricsSerivce` can be null. If it is null, there won't be any metrics data sent anywhere.

Reply:
Ya I think I added this by mistake, nice catch thanks